File: 01-Programming-Languages/PHP/Algorithms/Arrays/two_sum.php
<?php
function twoSum(array $nums, int $target): ?array {
    $numMap = [];

    foreach ($nums as $i => $num) {
        $complement = $target - $num;
        if (array_key_exists($complement, $numMap)) {
            return [$numMap[$complement], $i];
        }
        $numMap[$num] = $i;
    }

    return null;
}

$nums = [2, 7, 11, 15];
$target = 9;
$result = twoSum($nums, $target);

if ($result) {
    echo "Indices: [" . implode(", ", $result) . "]\n";
} else {
    echo "No solution found\n";
}
?>

File: 01-Programming-Languages/Ruby/Algorithms/Arrays/two_sum.rb
def two_sum(nums, target)
  num_map = {}

  nums.each_with_index do |num, i|
    complement = target - num
    if num_map.key?(complement)
      return [num_map[complement], i]
    end
    num_map[num] = i
  end

  nil
end

nums = [2, 7, 11, 15]
target = 9
result = two_sum(nums, target)
puts "Indices: #{result}"